k8s/deployments.go

```diff
@@ -98,60 +158,27 @@ func (c deployments) Create(appName, description string, file *runtime.File, sto
 		return nil, err
 	}
 	// check token...
-	if tk.IsAuthorized(*app.Team) == false {
+	if !tk.IsAuthorized(*app.Team) {
 		msg := "token not allowed to do a deployment"
 		return nil, NewUnauthorizedError(msg)
 	}
 	// creating deployment params
 	deploy := newDeploy(appName, description)
-	// create log context with uuid and app name
-	lc := log.WithField("app", appName).WithField("deployUUID", deploy.uuid)
-	lc.Info("starting deploy...")
 	// streaming actions...
 	r, w := io.Pipe()
+	done := make(chan struct{})
+	go c.create(w, app, file, deploy, storage, done)
 	go func() {
-		defer w.Close()
-
-		// parsing teresa.yaml and updating app
-		if err := parsingTeresaYaml(app, file.Data); err != nil {
-			lc.WithError(err).Error("error parsing teresa.yaml")
-			fmt.Fprintln(w, "error when searching and parsing teresa.yaml")
-			return
-		}
-
-		// upload file to storage
-		lcu := lc.WithField("storage", storage.Type()).WithField("storageIn", deploy.storageIn)
-		lcu.Debug("uploading app archive to storage...")
-		fmt.Fprintln(w, "uploading app archive to storage...")
-		// moving to the start of the file
-		file.Data.Seek(0, 0)
-		if err := storage.UploadFile(deploy.storageIn, file.Data); err != nil {
-			lcu.WithError(err).Error("error found when upload app archive to storage")
-			fmt.Fprintln(w, "error found when upload app archive to storage")
-			return
-		}
-		lcu.Debug("upload done with success")
-		fmt.Fprintln(w, "upload done with success")
-		lcu = nil
-		// building app...
-		lc.Debug("building app...")
-		fmt.Fprintln(w, "building app...")
-		if err := c.buildApp(app, deploy, storage, w); err != nil {
-			lc.WithError(err).Warn("error during build proccess")
-			fmt.Fprintln(w, "error during build proccess")
-			return
-		}
-		lc.Debug("build step done without errors")
-		fmt.Fprintln(w, "build step done without errors")
-		// updating deployment
-		lc.Debug("updating deployment for rolling update...")
-		fmt.Fprintln(w, "rolling update...")
-		if err := c.updateDeployment(app, deploy.slugPath, deploy.description, storage); err != nil {
-			lc.WithError(err).Error("error updating deployment")
-			fmt.Fprintln(w, "error when doing rolling update")
-			return
+	L:
+		for {
+			select {
+			case <-time.After(deploymentConfig.KeepAliveTimeout):
+				fmt.Fprint(w, keepAliveMessage)
+			case <-done:
+				break L
```

Comment:
or you can simplify it using two `break`, one for `select` and another for `for`.